Git-workflow.md

```diff
@@ -0,0 +1,147 @@
+# Git workflow
+---
+
+The standard git workflow we utilize is focused on maintaining a clean commit history for the benefit of outside observers, PR reviewers, and our future selves.
+
+The general steps for working on a new feature branch are as follows (commands covered in more detail below):
+
+
+### Setup
+Before using this workflow, perform the following setup steps:
+
+Block specific git commands:
+https://github.com/seattleflu/documentation/blob/master/Macbook-setup.md#block-specific-git-commands
+
+Install git-secrets:
+https://github.com/seattleflu/documentation/blob/master/Macbook-setup.md#install-git-secrets
+
+
+### Pre-PR:
+1. Checkout and pull latest main/master branch
+2. Create a new feature branch off of main/master
+3. Commit code changes at multiple points in development, with each commit representing a relatively discrete change that can be reviewed (and ideally, tested) independently. The commit history and associated comments should "tell a story" of the code changes, but omit details that are not relevant to the final version (e.g. bug fixes, typos).
+4. Update `.gitignore(s)` to account for any files that are being added/removed from tracking.
+5. Rebase feature branch to incorporate changes to master branch and/or to rework commit history on current branch
+6. Run automated tests locally (pytest, mypy, etc.)
+7. Check difference against remote branch before pushing
+8. Push code changes to Github
+9. Review feature branch commit history on Github
+10. Open PR when branch is ready for review
+
+### Post-PR review:
+11. Review feedback and make changes to code as needed
+12. Rebase and squash code changes into appropriate commits to result in a clean history
+13. (force) Push changes to Github, request additional review if needed.
+14. Merge and deploy (after PR approval)
+
+## Steps 1, 2 - Create a new feature branch
+```
+git checkout master
+git pull
+git checkout -b my-new-feature-branch
+```
+
+## Steps 3, 4 - Write code and commit changes
+(Note: Prior to running `git status`, manually update `.gitignore` files to add/remove files from tracking. Many of our repos include `.gitignore` files that function as allowlists as opposed to the standard denylist behavior)
+```
+git status
+git add [new or updated file]
+git commit
+```
+
+Commit comments generally should be written using these guidelines (https://cbea.ms/git-commit/).
+
+## Step 5 - Rebase to maintain clean history
+Interactive rebasing is a useful way to maitain a clean commit history. During an interactive rebase, commits can be merged, dropped, squashed; and comments added and edited.
```

Comment:
I use 'fixup' a lot, might be worth mentioning that here too.

Reply:
I haven't used that one before. Is that only when you're updating code by not comments? And is this how you'd use it?
```
git add <filename>
git commit --fixup <commit_id>
git rebase -i --autosquash
```